code/modules/recycling/disposal/bin.dm

```diff
@@ -100,8 +100,17 @@
 			return
 
 	if(!user.combat_mode)
+		var/was_latched_toolbox
+		var/obj/item/storage/toolbox/placed_toolbox
+		if(istype(I, /obj/item/storage/toolbox))
```

Comment:
This is a really bad hack and niche features like this *especially* should have really low surface area.

The best way to do this is to have these fire signals that the toolboxes then register to in order to know when to not dump all their items.

Reply:
I knew it wasnt the best way to do this, I coded it the best way I could and PR'd so people can look it over and help me figure out how to do it properly. Ill work on figuring out how signals work over the next few days